test/Tests/BpmSpeedTests.cs

```diff
@@ -136,21 +137,15 @@ public static void Rcv1Test(double wVariance, double biasVariance)
                 }
             }
             writer.Dispose();
-#if NETFRAMEWORK
-            // In the .NET 5.0 BinaryFormatter is obsolete
-            // and would produce errors. This test code should be migrated. 
-            // See https://aka.ms/binaryformatter
 
-            if (true)
+            using (Stream stream = File.Create(Path.Combine(dataFolder, "weights.bin")))
             {
-                BinaryFormatter serializer = new BinaryFormatter();
-                using (Stream stream = File.Create(Path.Combine(dataFolder, "weights.bin")))
-                {
-                    serializer.Serialize(stream, train.wPost);
-                    serializer.Serialize(stream, train.biasPost);
-                }
+                var serializer = SerializationUtils.GetJsonFormatter();
+#pragma warning disable SYSLIB0011
```

Comment:
IFormatter is marked as obsolete because they don't want people to use BinaryFormatter and that's what the error says. If we'd rather not have these then I can switch to a bespoke interface that is similar, or switch to the IReader/IWriter interfaces.

Reply:
We'd rather not have these.